euphoria-flink/src/main/java/cz/seznam/euphoria/flink/streaming/ReduceStateByKeyTranslator.java

```diff
@@ -73,7 +73,9 @@
     // assign windows
     DataStream<KeyedMultiWindowedElement> windowed = input.map(new WindowAssigner(
             windowing, keyExtractor, valueExtractor, eventTimeAssigner))
-            .setParallelism(operator.getParallelism());
+            // ~ execute in the same chain of the input's processing
+            // so far, thereby, avoiding an unnecessary shuffle
+            .setParallelism(input.getParallelism());
```

Comment:
* `input.getParallelism()` returns the parallelism the corresponding flink's dataset
* `operator.getParallism()` returns the _target_ parallelism intended for the operator (in this case, rsbk)